Use non deprecated method name

mathisto opened the pull request:
In prompt.rb:40 use Pry#input_ring vice Pry#input_array, which is being deprecated as per https://github.com/pry/pry/pull/1814

mathisto committed:
Use non deprecated method name

In prompt.rb:40 use Pry#input_ring vice Pry#input_array, which is being deprecated as per https://github.com/pry/pry/pull/1814

plribeiro3000 commented:
Cool. Thanks for the PR @mathisto.

What would happen if the user is locked in an old version of `pry`? would he still be able to bump `jazz_fingers`?

mathisto commented:
Awww shucks, you are right. This isn't backwards compatible fix. The `Pry::Ring` class was just added 10 days ago. It was previously History Array. The newest version of pry will throw a warning about the deprecation.
![image](https://user-images.githubusercontent.com/8226466/49544690-77ed4500-f8a9-11e8-883b-2f78fbf8efb2.png)
Based on the version of pry you have in your gemspec, you might want to just reject the PR, or hold onto it for when/if you update.

P.S. Ignore the fact I tried `ls` in the pry console. It a weird quirk of muscle memory.

plribeiro3000 commented:
@mathisto  believe we can work on top of what we already have here.

What if we check if the object respond to `input_ring` and based on it use one or the other?

Commit:
Add ternary decision for ring/array size

The newest version of pry renames the history array to ring. This selectively responds to the newest version of the method.

mathisto commented:
Great call with `respond_to?`. This works nicely and will support future changes to pry version.

Commit:
Replace multi line ternary

Taking the hounds advice. Multi-line ternaries are apparently frowned upon.

mathisto commented on lib/jazz_fingers/prompt.rb:
```diff
@@ -37,7 +37,9 @@ def name
 
     def line_number(pry)
-      "[#{bold_text(pry.input_array.size)}]"
+      pry.respond_to? :input_ring ?
```
Ok Hound, I'll bow to your demands.

plribeiro3000 commented:
LGTM! Thanks for bearing with me on this!
